dmutils/flask.py

```diff
@@ -32,3 +33,24 @@
 timed_render_template_string.__doc__ = """
     See ``timed_render_template``, only for ``render_template_string``.
 """
+
+
+class DMGzipMiddleware(Gzip):
+    compress_by_default: bool
+
+    def __init__(self, *args, compress_by_default: bool = False, **kwargs):
+        kwargs.setdefault("minimum_size", 8192)
+        self.compress_by_default = compress_by_default
+        super().__init__(*args, **kwargs)
+
+    def after_request(self, response):
+        x_compression_safe = response.headers.pop("X-Compression-Safe", None)
+        compress = {"0": False, "1": True}.get(x_compression_safe, self.compress_by_default)
```

Comment:
Why 0/1 instead of false/true?

Reply:
You don't tend to see false/true so much in header values, and then you also have all the uncertainty of `false`/`true`/`False`/`True`/`f`/`t`/`F`/`T`... when does it stop?